File: authorization/src/main/java/com/example/alexeyglushkov/authorization/OAuth/Token.java
package com.example.alexeyglushkov.authorization.OAuth;

import org.junit.Assert;

import java.io.Serializable;

import androidx.annotation.NonNull;

/**
 * Represents an OAuth token (either request or access token) and its secret
 *
 * @author Pablo Fernandez
 */
public class Token implements Serializable
{
  private static final long serialVersionUID = 715000866082812683L;

  private final String token;
  private final String secret;
  private final String rawResponse;

  /**
   * Default constructor
   *
   * @param token token value. Can't be null.
   * @param secret token secret. Can't be null.
   */
  public Token(String token, String secret) {
    this(token, secret, null);
  }

  public Token(String token, String secret, String rawResponse) {
    Assert.assertNotNull(token);
    Assert.assertNotNull(secret);

    this.token = token;
    this.secret = secret;
    this.rawResponse = rawResponse;
  }

  public String getToken() {
    return token;
  }

  public String getSecret() {
    return secret;
  }

  public String getRawResponse() {
    if (rawResponse == null) {
      throw new IllegalStateException("This token object was not constructed by scribe and does not have a rawResponse");
    }
    return rawResponse;
  }

  public String getParameter(String parameter) {
    String value = null;
    for (String str : this.getRawResponse().split("&"))
    {
      if (str.startsWith(parameter + '='))
      {
        String [] part = str.split("=");
        if (part.length > 1) {
          value = part[1].trim();
        }
        break;
      }
    }
    return value;
  }

  @Override @NonNull
  public String toString() {
    return String.format("Token[%s , %s]", token, secret);
  }

  /**
   * Returns true if the token is empty (token = "", secret = "")
   */
  public boolean isEmpty() {
    return "".equals(this.token) && "".equals(this.secret);
  }

  /**
   * Factory method that returns an empty token (token = "", secret = "").
   *
   * Useful for two legged OAuth.
   */
  public static Token empty() {
    return new Token("", "");
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;

    Token that = (Token) o;
    return token.equals(that.token) && secret.equals(that.secret);
  }

  @Override
  public int hashCode() {
    return 31 * token.hashCode() + 31 * secret.hashCode();
  }
}

File: authorization/src/main/java/com/example/alexeyglushkov/authorization/OAuth/Token.kt
package com.example.alexeyglushkov.authorization.OAuth

import org.junit.Assert
import java.io.Serializable

class Token @JvmOverloads constructor(token: String, secret: String, rawResponse: String? = null) : Serializable {
    val token: String
    val secret: String
    private val rawResponse: String?

    fun getRawResponse(): String {
        checkNotNull(rawResponse) { "This token object was not constructed by scribe and does not have a rawResponse" }
        return rawResponse
    }

    fun getParameter(parameter: String): String? {
        var value: String? = null
        for (str in getRawResponse().split("&").toTypedArray()) {
            if (str.startsWith("$parameter=")) {
                val part = str.split("=").toTypedArray()
                if (part.size > 1) {
                    value = part[1].trim { it <= ' ' }
                }
                break
            }
        }
        return value
    }

    override fun toString(): String {
        return String.format("Token[%s , %s]", token, secret)
    }

    /**
     * Returns true if the token is empty (token = "", secret = "")
     */
    val isEmpty: Boolean
        get() = "" == token && "" == secret

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as Token
        return token == that.token && secret == that.secret
    }

    override fun hashCode(): Int {
        return 31 * token.hashCode() + 31 * secret.hashCode()
    }

    companion object {
        private const val serialVersionUID = 715000866082812683L
        /**
         * Factory method that returns an empty token (token = "", secret = "").
         *
         * Useful for two legged OAuth.
         */
        fun empty(): Token {
            return Token("", "")
        }
    }

    /**
     * Default constructor
     *
     * @param token token value. Can't be null.
     * @param secret token secret. Can't be null.
     */
    init {
        Assert.assertNotNull(token)
        Assert.assertNotNull(secret)
        this.token = token
        this.secret = secret
        this.rawResponse = rawResponse
    }
}